src/components/metadata/metadata.test.js

```diff
@@ -93,70 +97,59 @@ describe('MetaData', () => {
     });
 
     it('shows the node type as an icon', () => {
-      const wrapper = mount({ nodeId: salmonTaskNodeId });
+      const wrapper = mount({
+        nodeId: salmonTaskNodeId,
+        mockMetadata: node_task,
+      });
       expect(rowIcon(wrapper).hasClass('pipeline-node-icon--icon-task')).toBe(
         true
       );
     });
 
     it('shows the node name as the title', () => {
-      const wrapper = mount({ nodeId: salmonTaskNodeId });
+      const wrapper = mount({
+        nodeId: salmonTaskNodeId,
+        mockMetadata: node_task,
+      });
       expect(textOf(title(wrapper))).toEqual(['salmon']);
     });
 
     it('shows the node type as text', () => {
-      const wrapper = mount({ nodeId: salmonTaskNodeId });
+      const wrapper = mount({
+        nodeId: salmonTaskNodeId,
+        mockMetadata: node_task,
+      });
       const row = rowByLabel(wrapper, 'Type:');
       expect(textOf(rowValue(row))).toEqual(['task']);
     });
 
     it('does not display the node parameter row when there are no parameters', () => {
-      const wrapper = mount({ nodeId: salmonTaskNodeId });
+      const wrapper = mount({
+        nodeId: salmonTaskNodeId,
+        mockMetadata: node_task,
+      });
       const row = rowByLabel(wrapper, 'Parameters:');
-      //this is output of react-json-view with no value
+      //this is the metadata output when there is no data
       expect(textOf(rowObject(row))).toEqual(['-']);
     });
 
     it('shows the node parameters when there are parameters', () => {
-      const metadata = getClickedNodeMetaData(
-        prepareState({
-          data: animals,
-          afterLayoutActions: [
-            () => toggleNodeClicked(salmonTaskNodeId),
-            () =>
-              addNodeMetadata({ id: salmonTaskNodeId, data: node_parameters }),
-          ],
-        })
-      );
-      const wrapper = setup.mount(
-        <MetaData visible={true} metadata={metadata} />
-      );
+      const wrapper = mount({
+        nodeId: salmonTaskNodeId,
+        mockMetadata: node_parameters,
+      });
       const row = rowByLabel(wrapper, 'Parameters:');
       //this is output of react-json-view with 3 parameters
       expect(textOf(rowObject(row))[0]).toEqual(
         expect.stringContaining('3 items')
       );
     });
 
-    it('shows the node inputs when parameters are disabled (default)', () => {
-      const wrapper = mount({ nodeId: salmonTaskNodeId });
-      const row = rowByLabel(wrapper, 'Inputs:');
-      expect(textOf(rowValue(row))).toEqual(['Cat', 'Dog']);
-    });
```

Comment:
I am confused as to why this test is deleted? Looking at the test it should still be valid amongst the changes and should still pass.

Reply:
So previously input/output were dependent on what nodes are hidden in the flowchart. So if parameters were disabled, the input/output would not show parameters and if they were enabled the input/output would list them. This has been changed in this ticket. We no longer read from the flowchart -- hence if parameters are disabled/enabled, the output would always be the same unlike before. This is the reason I removed the test.